File: OPENAPI-REST-API/swagger-client/dart/lib/model/server_order_bandwidth.dart
part of swagger.api;

class ServerOrderBandwidth {
  /* Bandwidth ID. */
  String id = null;
/* Bandwidth price. */
  int price = null;
/* Bandwidth image. */
  String img = null;
/* Short description of the bandwidth. */
  String shortDesc = null;
/* Long description of the bandwidth. */
  String longDesc = null;
/* Bandwidth type. */
  String type = null;
/* Quantity of bandwidth. */
  String qty = null;
/* Active status. */
  String active = null;
/* Monthly price. */
  int monthlyPrice = null;
/* Display of bandwidth price. */
  String priceDisplay = null;
/* Display of monthly bandwidth price. */
  String monthlyPriceDisplay = null;

  ServerOrderBandwidth();

  @override
  String toString() {
    return 'ServerOrderBandwidth[id=$id, price=$price, img=$img, shortDesc=$shortDesc, longDesc=$longDesc, type=$type, qty=$qty, active=$active, monthlyPrice=$monthlyPrice, priceDisplay=$priceDisplay, monthlyPriceDisplay=$monthlyPriceDisplay, ]';
  }

  ServerOrderBandwidth.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    id = json['id'];
    price = json['price'];
    img = json['img'];
    shortDesc = json['short_desc'];
    longDesc = json['long_desc'];
    type = json['type'];
    qty = json['qty'];
    active = json['active'];
    monthlyPrice = json['monthly_price'];
    priceDisplay = json['price_display'];
    monthlyPriceDisplay = json['monthly_price_display'];
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'price': price,
      'img': img,
      'short_desc': shortDesc,
      'long_desc': longDesc,
      'type': type,
      'qty': qty,
      'active': active,
      'monthly_price': monthlyPrice,
      'price_display': priceDisplay,
      'monthly_price_display': monthlyPriceDisplay
     };
  }

  static List<ServerOrderBandwidth> listFromJson(List<dynamic> json) {
    return json == null ? new List<ServerOrderBandwidth>() : json.map((value) => new ServerOrderBandwidth.fromJson(value)).toList();
  }

  static Map<String, ServerOrderBandwidth> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, ServerOrderBandwidth>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new ServerOrderBandwidth.fromJson(value));
    }
    return map;
  }
}

File: OPENAPI-REST-API/openapi-client/scalaz/src/main/scala/org/openapitools/client/api/ServerOrderBandwidth.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ServerOrderBandwidth._

case class ServerOrderBandwidth (
  /* Bandwidth ID. */
  id: Option[String],
/* Bandwidth price. */
  price: Option[Integer],
/* Bandwidth image. */
  img: Option[String],
/* Short description of the bandwidth. */
  short_desc: Option[String],
/* Long description of the bandwidth. */
  long_desc: Option[String],
/* Bandwidth type. */
  `type`: Option[String],
/* Quantity of bandwidth. */
  qty: Option[String],
/* Active status. */
  active: Option[String],
/* Monthly price. */
  monthly_price: Option[Integer],
/* Display of bandwidth price. */
  price_display: Option[String],
/* Display of monthly bandwidth price. */
  monthly_price_display: Option[String])

object ServerOrderBandwidth {
  import DateTimeCodecs._

  implicit val ServerOrderBandwidthCodecJson: CodecJson[ServerOrderBandwidth] = CodecJson.derive[ServerOrderBandwidth]
  implicit val ServerOrderBandwidthDecoder: EntityDecoder[ServerOrderBandwidth] = jsonOf[ServerOrderBandwidth]
  implicit val ServerOrderBandwidthEncoder: EntityEncoder[ServerOrderBandwidth] = jsonEncoderOf[ServerOrderBandwidth]
}
